src/Pos/Context/Context.hs

```diff
@@ -108,6 +109,8 @@ data NodeContext ssc = NodeContext
     -- ^ Context needed for the update system
     , ncLrcContext          :: !LrcContext
     -- ^ Context needed for LRC
+    , ncDiscoveryContext    :: !DiscoveryContextSum
```

Comment:
1. It's used by `DiscoveryRedirect` which needs `Ether.MonadReader' DiscoveryContextSum` for `MonadDiscovery` instance.

2. Yes, it can be.